#### Importing pandas

In [1]:
import pandas as pd
import numpy as np

#### Importing the data

In [2]:
geo=pd.read_csv(r"C:\Users\USER\Downloads\cells_geo.csv")
day1=pd.read_csv(r"C:\Users\USER\Downloads\Telcom_dataset.csv")
day2=pd.read_csv(r"C:\Users\USER\Downloads\Telcom_dataset2.csv")
day3=pd.read_csv(r"C:\Users\USER\Downloads\Telcom_dataset3.csv")
geo_description=pd.read_excel(r"C:\Users\USER\Downloads\cells_geo_description.xlsx")
CDR_description=pd.read_excel(r"C:\Users\USER\Downloads\CDR_description.xlsx")

#### Exploring the datasets

##### 1. Geo

In [3]:
geo.head()

;VILLES;STATUS;LOCALISATION;DECOUPZONE;ZONENAME;LONGITUDE;LATITUDE;REGION;AREA;CELL_ID;SITE_CODE
0  0;ADJAME;In Service;ABIDJAN;"""Abidjan_EST";AG...                                              
1  1;ADJAME;In Service;ABIDJAN;"""Abidjan_EST";AG...                                              
2  2;ADJAME;In Service;ABIDJAN;"""Abidjan_EST";AG...                                              
3  3;ASSINIE;In Service;INTERIEUR;Grand-EST;"ASSI...                                              
4  4;ASSINIE;In Service;INTERIEUR;Grand-EST;"ASSI...

In [4]:
geo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3974 entries, 0 to 3973
Data columns (total 1 columns):
 #   Column                                                                                            Non-Null Count  Dtype 
---  ------                                                                                            --------------  ----- 
 0   ;VILLES;STATUS;LOCALISATION;DECOUPZONE;ZONENAME;LONGITUDE;LATITUDE;REGION;AREA;CELL_ID;SITE_CODE  3974 non-null   object
dtypes: object(1)
memory usage: 31.2+ KB


**Separating the columns**

In [5]:
geo=geo[";VILLES;STATUS;LOCALISATION;DECOUPZONE;ZONENAME;LONGITUDE;LATITUDE;REGION;AREA;CELL_ID;SITE_CODE"].str.split(';',
                                                                                                                      expand=True)

In [6]:
geo.head()

0        1           2          3                4            5   \
0  0   ADJAME  In Service    ABIDJAN  """Abidjan_EST"     AGBANNEW   
1  1   ADJAME  In Service    ABIDJAN  """Abidjan_EST"     AGBANNEW   
2  2   ADJAME  In Service    ABIDJAN  """Abidjan_EST"     AGBANNEW   
3  3  ASSINIE  In Service  INTERIEUR        Grand-EST  "ASSINIE"""   
4  4  ASSINIE  In Service  INTERIEUR        Grand-EST  "ASSINIE"""   

                  6                 7          8  9           10          11  
0  -4.00066666666667  5.36819444444444     LAGUNE     fbb2e75af3  0bb492ef17  
1  -4.00066666666667  5.36819444444444     LAGUNE     1502501d78  0bb492ef17  
2  -4.00066666666667  5.36819444444444     LAGUNE     0c2e182a0c  0bb492ef17  
3  -3.48846666666667  5.15876111111111  SUD COMOE     69bdfc69fd  8a1e6ce31f  
4  -3.48846666666667  5.15876111111111  SUD COMOE     7a1029d2ab  8a1e6ce31f

**Renaming the columns correctly**

In [7]:
geo=geo.rename(columns={1:"VILLES",
                    2:"STATUS",
                    3:"LOCALISATION",
                    4:"DECOUPZONE",
                    5:"ZONENAME",
                    6:"LONGITUDE",
                    7:"LATITUDE",
                    8:"REGION",
                    9:"AREA",
                    10:"CELL_ID",
                    11:"SITE_CODE"
                  })

In [8]:
geo.drop(0,inplace=True, axis=1)

In [9]:
geo.head()

VILLES      STATUS LOCALISATION       DECOUPZONE     ZONENAME  \
0   ADJAME  In Service      ABIDJAN  """Abidjan_EST"     AGBANNEW   
1   ADJAME  In Service      ABIDJAN  """Abidjan_EST"     AGBANNEW   
2   ADJAME  In Service      ABIDJAN  """Abidjan_EST"     AGBANNEW   
3  ASSINIE  In Service    INTERIEUR        Grand-EST  "ASSINIE"""   
4  ASSINIE  In Service    INTERIEUR        Grand-EST  "ASSINIE"""   

           LONGITUDE          LATITUDE     REGION AREA     CELL_ID   SITE_CODE  
0  -4.00066666666667  5.36819444444444     LAGUNE       fbb2e75af3  0bb492ef17  
1  -4.00066666666667  5.36819444444444     LAGUNE       1502501d78  0bb492ef17  
2  -4.00066666666667  5.36819444444444     LAGUNE       0c2e182a0c  0bb492ef17  
3  -3.48846666666667  5.15876111111111  SUD COMOE       69bdfc69fd  8a1e6ce31f  
4  -3.48846666666667  5.15876111111111  SUD COMOE       7a1029d2ab  8a1e6ce31f

**Summary of the Geo data**

In [10]:
geo_description

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [11]:
geo.describe(include="all")

VILLES      STATUS LOCALISATION   DECOUPZONE              ZONENAME  \
count     3974        3974         3974         3974                  3974   
unique     504           2            2           15                    34   
top     COCODY  In Service    INTERIEUR  Grand-OUEST  ZONE D - Grand-OUEST   
freq       354        3907         2305          528                   527   

                LONGITUDE          LATITUDE  REGION           AREA  \
count                3974              3974    3974           3974   
unique               1013              1025      19              7   
top     -4.02333333333333  5.37363888888889  LAGUNE  ABIDJAN NORTH   
freq                   12                12    1850           1222   

           CELL_ID   SITE_CODE  
count         3974        3974  
unique        3916        1029  
top     6750e0fb7b  20387d4292  
freq            39          42

In [12]:
geo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3974 entries, 0 to 3973
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   VILLES        3974 non-null   object
 1   STATUS        3974 non-null   object
 2   LOCALISATION  3974 non-null   object
 3   DECOUPZONE    3974 non-null   object
 4   ZONENAME      3974 non-null   object
 5   LONGITUDE     3974 non-null   object
 6   LATITUDE      3974 non-null   object
 7   REGION        3974 non-null   object
 8   AREA          3974 non-null   object
 9   CELL_ID       3974 non-null   object
 10  SITE_CODE     3974 non-null   object
dtypes: object(11)
memory usage: 341.6+ KB


- There are no null values because every row is in string format

In [13]:
geo["DECOUPZONE"].unique()

array(['"""Abidjan_EST"', 'Grand-EST', 'Grand-OUEST', '"""Abidjan_NORD"',
       '"""Abidjan_CENTRE"', 'Abidjan_NORD', 'Abidjan_EST',
       'Abidjan_OUEST', 'Grand-SUD', 'SUD-OUEST', 'Abidjan_CENTRE',
       'Abidjan_SUD', 'Grand-NORD', 'Gd-ABIDJAN', '0'], dtype=object)

In [14]:
geo.replace('"""Abidjan_EST"', "Abidjan_EST",inplace=True)
geo.replace('"""Abidjan_CENTRE"', "Abidjan_CENTRE",inplace=True)
geo.replace('"""Abidjan_NORD"', "Abidjan_NORD",inplace=True)

In [15]:
geo["ZONENAME"].unique()

array(['AGBANNEW', '"ASSINIE"""', '"KRIKOREA"""', 'NPONON', 'OKAHOTEL',
       'PALMABINGERVILLE', 'TIAMA', 'WILLIAMSVILLE2',
       'ZONE E - Abidjan_NORD', 'ZONE B - Abidjan_EST',
       'ZONE D - Grand-OUEST', 'ZONE C - Grand-EST',
       'ZONE C - Abidjan_OUEST', 'ZONE B - Grand-SUD',
       'ZONE E - SUD-OUEST', 'ZONE A - Abidjan_CENTRE',
       'ZONE B - Abidjan_OUEST', 'ZONE B - Abidjan_CENTRE',
       'ZONE D - Abidjan_SUD', 'ZONE B - Grand-EST',
       'ZONE A - Grand-NORD', 'ZONE D - Grand-SUD',
       'ZONE D - Abidjan_EST', 'ZONE C - Grand-SUD',
       'ZONE C - Abidjan_SUD', 'ZONE A - Grand-SUD',
       'ZONE C - Abidjan_EST', 'ZONE B - Abidjan_Nord', '',
       'ZONE A - Grand-EST', 'ZONE B - Abidjan-EST',
       'ZONE A - Abidjan_OUEST', 'ZONE B - Abidjan_SUD',
       'ZONE B - Grand-NORD'], dtype=object)

In [16]:
geo.replace('"ASSINIE"""', "ASSINIE",inplace=True)
geo.replace('"KRIKOREA"""', "KRIKOREA",inplace=True)

In [17]:
geo["STATUS"].unique()

array(['In Service', ''], dtype=object)

In [18]:
geo["STATUS"].replace("","Not In Service", inplace=True)

In [19]:
geo["STATUS"].unique()

array(['In Service', 'Not In Service'], dtype=object)

In [20]:
geo["LOCALISATION"].unique()

array(['ABIDJAN', 'INTERIEUR'], dtype=object)

In [21]:
geo["REGION"].unique()

array(['LAGUNE', 'SUD COMOE', 'HAUT SASSANDRA', 'MOYEN COMOE', 'AGNEBY',
       'SUD BANDAMA', 'BAS SASSANDRA', 'LACS', 'FROMAGER',
       'VALLEE DU BANDAMA', 'NZI COMOE', 'ZANZAN', 'MOYEN CAVALLY',
       'WORODOUGOU', 'DENGUELE', '18 MONTAGNES', 'SAVANE', 'MARAHOUE',
       'BAFING'], dtype=object)

In [22]:
geo["AREA"].unique()

array(['', 'ABIDJAN NORTH', 'NORTH WEST', 'EAST', 'ABIDJAN SOUTH',
       'SOUTH WEST', 'CENTRE NORTH'], dtype=object)

In [23]:
geo["AREA"].replace('',np.nan,inplace=True)

In [24]:
geo["AREA"].unique()

array([nan, 'ABIDJAN NORTH', 'NORTH WEST', 'EAST', 'ABIDJAN SOUTH',
       'SOUTH WEST', 'CENTRE NORTH'], dtype=object)

In [25]:
geo["LONGITUDE"]=geo["LONGITUDE"].astype(float)
geo["LATITUDE"]=geo["LATITUDE"].astype(float)

In [26]:
geo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3974 entries, 0 to 3973
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   VILLES        3974 non-null   object 
 1   STATUS        3974 non-null   object 
 2   LOCALISATION  3974 non-null   object 
 3   DECOUPZONE    3974 non-null   object 
 4   ZONENAME      3974 non-null   object 
 5   LONGITUDE     3974 non-null   float64
 6   LATITUDE      3974 non-null   float64
 7   REGION        3974 non-null   object 
 8   AREA          3951 non-null   object 
 9   CELL_ID       3974 non-null   object 
 10  SITE_CODE     3974 non-null   object 
dtypes: float64(2), object(9)
memory usage: 341.6+ KB


In [27]:
geo.isnull().sum()

VILLES           0
STATUS           0
LOCALISATION     0
DECOUPZONE       0
ZONENAME         0
LONGITUDE        0
LATITUDE         0
REGION           0
AREA            23
CELL_ID          0
SITE_CODE        0
dtype: int64

- Note that there are now 23 null values in the Area column

##### 2. Day 1

In [28]:
day1.describe(include="all")

PRODUTC        VALUE               DATETIME CELL_ON_SITE  \
count     5001  5001.000000                   5001         5001   
unique       3          NaN                    708            7   
top      Voice          NaN  2012-05-07 00:00:10.0            1   
freq      2219          NaN                     71         1252   
mean       NaN    25.869626                    NaN          NaN   
std        NaN   101.934502                    NaN          NaN   
min        NaN     0.000000                    NaN          NaN   
25%        NaN     0.000000                    NaN          NaN   
50%        NaN     0.000000                    NaN          NaN   
75%        NaN    25.000000                    NaN          NaN   
max        NaN  4440.000000                    NaN          NaN   

        DW_A_NUMBER_INT  DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID  \
count              5001             5001      5001      5001        5001   
unique             3903             2848         1         2        1790   
top     4e49073a756df8c  5a4dc9e82deb309     19e2e     19e2e  ffa6759bb2   
freq                 63              807      5001      4026         716   
mean                NaN              NaN       NaN       NaN         NaN   
std                 NaN              NaN       NaN       NaN         NaN   
min                 NaN              NaN       NaN       NaN         NaN   
25%                 NaN              NaN       NaN       NaN         NaN   
50%                 NaN              NaN       NaN       NaN         NaN   
75%                 NaN              NaN       NaN       NaN         NaN   
max                 NaN              NaN       NaN       NaN         NaN   

           SITE_ID  
count         4285  
unique        1076  
top     b47a998a3e  
freq            64  
mean           NaN  
std            NaN  
min            NaN  
25%            NaN  
50%            NaN  
75%            NaN  
max            NaN

- There are outliers in the value column. The mean is 25.869626	but the minimum value is 0 and the 75th percentile is 25 yet the maximum value is 4440

In [29]:
day1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5001 entries, 0 to 5000
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   PRODUTC          5001 non-null   object
 1   VALUE            5001 non-null   int64 
 2   DATETIME         5001 non-null   object
 3   CELL_ON_SITE     5001 non-null   object
 4   DW_A_NUMBER_INT  5001 non-null   object
 5   DW_B_NUMBER_INT  5001 non-null   object
 6   COUNTRY_A        5001 non-null   object
 7   COUNTRY_B        5001 non-null   object
 8   CELL_ID          5001 non-null   object
 9   SITE_ID          4285 non-null   object
dtypes: int64(1), object(9)
memory usage: 390.8+ KB


In [30]:
day1.isnull().sum()

PRODUTC              0
VALUE                0
DATETIME             0
CELL_ON_SITE         0
DW_A_NUMBER_INT      0
DW_B_NUMBER_INT      0
COUNTRY_A            0
COUNTRY_B            0
CELL_ID              0
SITE_ID            716
dtype: int64

- There are 716 null values in the site ID column


**Renaming column 1**

In [31]:
day1.rename(columns={"PRODUTC":"PRODUCT"},inplace=True)

In [32]:
day1.head()

PRODUCT  VALUE               DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0   Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1     sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2     sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3   Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4   Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   

   DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0  1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1  93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2  7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3  6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4  02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv

In [33]:
head=day1.head(10)
tail=day1.tail(10)
frames2=[head,tail]
head_tail=pd.concat(frames2)
head_tail

PRODUCT  VALUE               DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
5       data      0  2012-05-06 23:07:26.0            _  f71825ca6e623f9   
6      Voice      0  2012-05-06 23:08:02.0            1  4afd7d5af21b1ce   
7      Voice      0  2012-05-06 23:08:03.0            5  d717dee81c51389   
8      Voice      0  2012-05-06 23:08:05.0            1  44522aefdc918a6   
9        sms      0  2012-05-06 23:08:43.0            0  0763a571d24db64   
4991     sms      0  2012-05-07 00:01:54.0            1  654993a4ff3fbf5   
4992     sms     50  2012-05-07 00:01:54.0            0  3ac714c9b3dadf4   
4993     sms     50  2012-05-07 00:01:54.0            2  79886d8e80f2fdc   
4994     sms      0  2012-05-07 00:01:54.0            1  5c791fc23361b5a   
4995     sms      0  2012-05-07 00:01:54.0            1  fb2907bd1331a92   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

      DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
5     5a4dc9e82deb309     19e2e     OTHER  /lrUXVGVcN  /lrUXVGVcN  
6     67bf1ac30e58e2f     19e2e     19e2e  +38u2u/rfx  +38u2u/rfx  
7     adde5afd9a67b10     19e2e     19e2e  09UjmmY1Ds  09UjmmY1Ds  
8     19e68e80103885e     19e2e     19e2e  0DeimCpZOc  0DeimCpZOc  
9     f773694c00c74d2     19e2e     19e2e  01MDiY5F89  01MDiY5F89  
4991  6a641b6f304a541     19e2e     19e2e  85bbd0e6b0  481aaf0c3b  
4992  a55a693a56941cf     19e2e     19e2e  6f3a2df445  f939f3349c  
4993  92bdd7f50c71071     19e2e     19e2e  babc0339bd  13c690c436  
4994  aaaf0e4bdec7311     19e2e     19e2e  0dc2b1dafc  ec013a6f3b  
4995  28b9dd8f413f87b     19e2e     19e2e  5116d1c4a6  c07a544c63  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70

#### Changing the datetime column from string to datetime format, and introducing hour column

In [34]:
import datetime

In [35]:
day1_new=day1.copy()


In [36]:
day1_new["DATETIME"]=pd.to_datetime(day1_new["DATETIME"]).dt.to_period("d")

In [37]:
day1_new.head()

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  DW_B_NUMBER_INT  \
0   Voice      0  2012-05-06            1  49fae048b547780  1a49f8b0ded9de6   
1     sms      0  2012-05-06            0  7a54a6eaee19e65  93343ba6e41af95   
2     sms     25  2012-05-06            0  6008887b0207d82  7aacc9709ce0a20   
3   Voice     56  2012-05-06            1  7370032e20f4e96  6af4a60e74c91d3   
4   Voice     14  2012-05-06            1  966ba4656e38e62  02aa736ee555d5c   

  COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     19e2e     19e2e  051KhYDCpv  051KhYDCpv

In [38]:
day1_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5001 entries, 0 to 5000
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype    
---  ------           --------------  -----    
 0   PRODUCT          5001 non-null   object   
 1   VALUE            5001 non-null   int64    
 2   DATETIME         5001 non-null   period[D]
 3   CELL_ON_SITE     5001 non-null   object   
 4   DW_A_NUMBER_INT  5001 non-null   object   
 5   DW_B_NUMBER_INT  5001 non-null   object   
 6   COUNTRY_A        5001 non-null   object   
 7   COUNTRY_B        5001 non-null   object   
 8   CELL_ID          5001 non-null   object   
 9   SITE_ID          4285 non-null   object   
dtypes: int64(1), object(8), period[D](1)
memory usage: 390.8+ KB


In [39]:
def hr_function(ts):
    return ts.hour

In [40]:
day1["DATETIME"]=pd.to_datetime(day1["DATETIME"])

In [41]:
day1_new["Hour"]=day1["DATETIME"].apply(hr_function)

In [42]:
day1_new.head()

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  DW_B_NUMBER_INT  \
0   Voice      0  2012-05-06            1  49fae048b547780  1a49f8b0ded9de6   
1     sms      0  2012-05-06            0  7a54a6eaee19e65  93343ba6e41af95   
2     sms     25  2012-05-06            0  6008887b0207d82  7aacc9709ce0a20   
3   Voice     56  2012-05-06            1  7370032e20f4e96  6af4a60e74c91d3   
4   Voice     14  2012-05-06            1  966ba4656e38e62  02aa736ee555d5c   

  COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  Hour  
0     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu    23  
1     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41    23  
2     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi    23  
3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3    23  
4     19e2e     19e2e  051KhYDCpv  051KhYDCpv    23

In [43]:
day1_new["Hour"].unique()

array([23,  0], dtype=int64)

##### Day 2

In [44]:
day2.describe(include="all")

PRODUCT        VALUE              DATE_TIME CELL_ON_SITE  \
count     5001  5001.000000                   5001         5001   
unique       3          NaN                    788            8   
top        sms          NaN  2012-05-08 00:00:10.0            1   
freq      2329          NaN                     82         1240   
mean       NaN    24.334933                    NaN          NaN   
std        NaN    97.471586                    NaN          NaN   
min        NaN     0.000000                    NaN          NaN   
25%        NaN     0.000000                    NaN          NaN   
50%        NaN     0.000000                    NaN          NaN   
75%        NaN    25.000000                    NaN          NaN   
max        NaN  3380.000000                    NaN          NaN   

            DW_A_NUMBER      DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID  \
count              5001             5001      5001      5001        5001   
unique             3799             2882         1         2        1661   
top     8ab8b9ed6989f61  5a4dc9e82deb309     19e2e     19e2e  ffa6759bb2   
freq                 31              766      5001      4047         676   
mean                NaN              NaN       NaN       NaN         NaN   
std                 NaN              NaN       NaN       NaN         NaN   
min                 NaN              NaN       NaN       NaN         NaN   
25%                 NaN              NaN       NaN       NaN         NaN   
50%                 NaN              NaN       NaN       NaN         NaN   
75%                 NaN              NaN       NaN       NaN         NaN   
max                 NaN              NaN       NaN       NaN         NaN   

           SITE_ID  
count         4325  
unique        1027  
top     91e86be9a6  
freq            57  
mean           NaN  
std            NaN  
min            NaN  
25%            NaN  
50%            NaN  
75%            NaN  
max            NaN

- There are outliers in the value column. The mean is 24.334933	but the minimum value is 0 and the 75th percentile is 25, yet the maximum value is 3380.

- There are null values in the site ID column

**Renaming columns**

In [45]:
day2.rename(columns={"DATE_TIME":"DATETIME"},inplace=True)
day2.rename(columns={"DW_A_NUMBER":"DW_A_NUMBER_INT"},inplace=True)
day2.rename(columns={"DW_B_NUMBER":"DW_B_NUMBER_INT"},inplace=True)

#### Changing the datetime column from string to datetime format, and introducing hour column

In [46]:
day2["DATETIME"]=pd.to_datetime(day2["DATETIME"])

In [47]:
day2_new=day2.copy()

In [48]:
day2_new["DATETIME"]=pd.to_datetime(day2_new["DATETIME"]).dt.to_period("d")

In [49]:
day2_new["Hour"]=day2["DATETIME"].apply(hr_function)

##### Day 3

In [50]:
day3.describe(include="all")

PRODUCT         VALUE              DATE_TIME CELL_ON_SITE  \
count     5001   5001.000000                   5001         5001   
unique       3           NaN                    647            9   
top        sms           NaN  2012-05-09 00:00:10.0            1   
freq      2767           NaN                     87         1239   
mean       NaN     26.441512                    NaN          NaN   
std        NaN    224.200217                    NaN          NaN   
min        NaN      0.000000                    NaN          NaN   
25%        NaN      0.000000                    NaN          NaN   
50%        NaN      0.000000                    NaN          NaN   
75%        NaN     25.000000                    NaN          NaN   
max        NaN  12900.000000                    NaN          NaN   

        DW_A_NUMBER_INT  DW_B_NUMBER_INT COUNTRY_A COUNTRY_B      CELLID  \
count              5001             5001      5001      5001        5001   
unique             3593             2895         1         2        1596   
top     8ab8b9ed6989f61  5a4dc9e82deb309     19e2e     19e2e  ffa6759bb2   
freq                 32              693      5001      4229         604   
mean                NaN              NaN       NaN       NaN         NaN   
std                 NaN              NaN       NaN       NaN         NaN   
min                 NaN              NaN       NaN       NaN         NaN   
25%                 NaN              NaN       NaN       NaN         NaN   
50%                 NaN              NaN       NaN       NaN         NaN   
75%                 NaN              NaN       NaN       NaN         NaN   
max                 NaN              NaN       NaN       NaN         NaN   

           SIET_ID  
count         4396  
unique         961  
top     1ddea341cd  
freq            48  
mean           NaN  
std            NaN  
min            NaN  
25%            NaN  
50%            NaN  
75%            NaN  
max            NaN

- There are outliers in the value column. The mean is 26.441512	but the minimum value is 0 and the 75th percentile is 25, yet the maximum value is 12900.

- There are null values in the site ID column


**Renaming the columns**

In [51]:
day3.rename(columns={"SIET_ID":"SITE_ID"},inplace=True)
day3.rename(columns={"DATE_TIME":"DATETIME"},inplace=True)
day3.rename(columns={"CELLID":"CELL_ID"},inplace=True)

#### Changing the datetime column from string to datetime format, and introducing hour column

In [52]:
day3["DATETIME"]=pd.to_datetime(day3["DATETIME"])

In [53]:
day3_new=day3.copy()

In [54]:
day3_new["DATETIME"]=pd.to_datetime(day3_new["DATETIME"]).dt.to_period("d")

In [55]:
day3_new["Hour"]=day3["DATETIME"].apply(hr_function)

#### More exploration

**_Average billing price per day per product_**

In [56]:
day1_new.groupby("Hour")["VALUE"].mean()

Hour
0     26.863102
23    19.358006
Name: VALUE, dtype: float64

In [57]:
day1.groupby("PRODUCT")["VALUE"].mean()

PRODUCT
Voice    49.062190
data      8.713755
sms       6.821772
Name: VALUE, dtype: float64

In [58]:
day2_new.groupby("Hour")["VALUE"].mean()

Hour
0     25.949941
23    15.406005
Name: VALUE, dtype: float64

In [59]:
day2.groupby("PRODUCT")["VALUE"].mean()

PRODUCT
Voice    54.294858
data      7.180157
sms       5.458566
Name: VALUE, dtype: float64

In [60]:
day3.groupby("PRODUCT")["VALUE"].mean()

PRODUCT
Voice    65.047372
data      6.861472
sms       9.844958
Name: VALUE, dtype: float64

**_Total billing price per day per product_**

In [61]:
day1.groupby("PRODUCT")["VALUE"].sum()

PRODUCT
Voice    108869
data       7032
sms       13473
Name: VALUE, dtype: int64

In [62]:
day2.groupby("PRODUCT")["VALUE"].sum()

PRODUCT
Voice    103486
data       5500
sms       12713
Name: VALUE, dtype: int64

In [63]:
day3.groupby("PRODUCT")["VALUE"].sum()

PRODUCT
Voice    100238
data       4755
sms       27241
Name: VALUE, dtype: int64

#### Joining the dataframes

In [64]:
frames=[day1_new,day2_new,day3_new]
all_days=pd.concat(frames,ignore_index=True)

In [65]:
all_days.head()

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  DW_B_NUMBER_INT  \
0   Voice      0  2012-05-06            1  49fae048b547780  1a49f8b0ded9de6   
1     sms      0  2012-05-06            0  7a54a6eaee19e65  93343ba6e41af95   
2     sms     25  2012-05-06            0  6008887b0207d82  7aacc9709ce0a20   
3   Voice     56  2012-05-06            1  7370032e20f4e96  6af4a60e74c91d3   
4   Voice     14  2012-05-06            1  966ba4656e38e62  02aa736ee555d5c   

  COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  Hour  
0     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu    23  
1     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41    23  
2     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi    23  
3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3    23  
4     19e2e     19e2e  051KhYDCpv  051KhYDCpv    23

In [66]:
all_days.tail()

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
14998   Voice      0  2012-05-09            3  467333ade2b4b9c   
14999   Voice     62  2012-05-09            1  e077a6b74c5053c   
15000   Voice     32  2012-05-09            1  77f53c39d27c050   
15001   Voice     30  2012-05-09            2  672ce74c418aa7a   
15002   Voice     60  2012-05-09            2  6edeb4c55714650   

       DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  Hour  
14998  0ab2acd804d01fe     19e2e     19e2e  2aa04dcab7  f7d63a29e5     0  
14999  4c8098eb6ea1a41     19e2e     19e2e  7f1f2f889f  b4ddb9e93c     0  
15000  9d064f013e2cab2     19e2e     19e2e  a4731e780e  d68260ef50     0  
15001  f9ff5af59cf0d68     19e2e     19e2e  98ed726da2  b4130b54ab     0  
15002  7e60838216482a7     19e2e     19e2e  1c11a2fcbb  6d70dae094     0

In [67]:
all_days.shape

(15003, 11)

##### Average billing price per product over the three days

In [68]:
all_days.groupby("PRODUCT")["VALUE"].mean()

PRODUCT
Voice    55.169961
data      7.628861
sms       7.555791
Name: VALUE, dtype: float64

##### Total billing price per product over the three days

In [69]:
all_days.groupby("PRODUCT")["VALUE"].sum()

PRODUCT
Voice    312593
data      17287
sms       53427
Name: VALUE, dtype: int64

### Inner and left join the geo  dataframe

In [70]:
geo.rename(columns={"SITE_CODE":"SITE_ID"},inplace=True)

**_Inner join by the site ID column gets rid of the null SITE ID values_**

In [71]:
CDR_description

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

In [72]:
geo2=geo.drop("CELL_ID",axis=1)

In [73]:
geo2.head()

VILLES      STATUS LOCALISATION   DECOUPZONE  ZONENAME  LONGITUDE  \
0   ADJAME  In Service      ABIDJAN  Abidjan_EST  AGBANNEW  -4.000667   
1   ADJAME  In Service      ABIDJAN  Abidjan_EST  AGBANNEW  -4.000667   
2   ADJAME  In Service      ABIDJAN  Abidjan_EST  AGBANNEW  -4.000667   
3  ASSINIE  In Service    INTERIEUR    Grand-EST   ASSINIE  -3.488467   
4  ASSINIE  In Service    INTERIEUR    Grand-EST   ASSINIE  -3.488467   

   LATITUDE     REGION AREA     SITE_ID  
0  5.368194     LAGUNE  NaN  0bb492ef17  
1  5.368194     LAGUNE  NaN  0bb492ef17  
2  5.368194     LAGUNE  NaN  0bb492ef17  
3  5.158761  SUD COMOE  NaN  8a1e6ce31f  
4  5.158761  SUD COMOE  NaN  8a1e6ce31f

In [74]:
all_days_2=all_days.drop("CELL_ID",axis=1)

In [75]:
joined_inner=pd.merge(all_days_2,
                      geo2,
                     how="inner",
                     on="SITE_ID")

In [76]:
head=joined_inner.head(10)
tail=joined_inner.tail(10)
frames2=[head,tail]
head_tail=pd.concat(frames2)
head_tail

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0         sms      0  2012-05-07            1  8118c3aa53d6c7b   
1         sms      0  2012-05-07            1  8118c3aa53d6c7b   
2         sms      0  2012-05-07            1  8118c3aa53d6c7b   
3         sms      0  2012-05-07            1  8118c3aa53d6c7b   
4         sms      0  2012-05-07            1  8118c3aa53d6c7b   
5         sms      0  2012-05-07            1  8118c3aa53d6c7b   
6       Voice     92  2012-05-07            4  ada82842ff7cf09   
7       Voice     92  2012-05-07            4  ada82842ff7cf09   
8       Voice     92  2012-05-07            4  ada82842ff7cf09   
9       Voice     92  2012-05-07            4  ada82842ff7cf09   
51351     sms      0  2012-05-09            1  dd953c2ab18c1b7   
51352     sms      0  2012-05-09            1  dd953c2ab18c1b7   
51353   Voice     47  2012-05-09            1  144d260d1961913   
51354   Voice     47  2012-05-09            1  144d260d1961913   
51355   Voice     47  2012-05-09            1  144d260d1961913   
51356   Voice     17  2012-05-09            4  22e493a92442b73   
51357   Voice     17  2012-05-09            4  22e493a92442b73   
51358     sms      0  2012-05-09            2  32e2e7c90a9b0e9   
51359     sms      0  2012-05-09            2  32e2e7c90a9b0e9   
51360     sms      0  2012-05-09            2  32e2e7c90a9b0e9   

       DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     SITE_ID  Hour   VILLES  \
0      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
1      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
2      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
3      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
4      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
5      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   BOUAKE   
6      d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   BOUAKE   
7      d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   BOUAKE   
8      d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   BOUAKE   
9      d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   BOUAKE   
51351  4b30ffe72e8a405     19e2e     19e2e  d236ec6ca8     0  TIAPOUM   
51352  4b30ffe72e8a405     19e2e     19e2e  d236ec6ca8     0  TIAPOUM   
51353  fc15d238a9d904f     19e2e     19e2e  aa59cc7ced     0  BLANFLA   
51354  fc15d238a9d904f     19e2e     19e2e  aa59cc7ced     0  BLANFLA   
51355  fc15d238a9d904f     19e2e     19e2e  aa59cc7ced     0  BLANFLA   
51356  c9e2c9c2fa3795a     19e2e     19e2e  cc45940406     0   ADJAME   
51357  c9e2c9c2fa3795a     19e2e     19e2e  cc45940406     0   ADJAME   
51358  32e2e7c90a9b0e9     19e2e     19e2e  ab91f5b21d     0    DIABO   
51359  32e2e7c90a9b0e9     19e2e     19e2e  ab91f5b21d     0    DIABO   
51360  32e2e7c90a9b0e9     19e2e     19e2e  ab91f5b21d     0    DIABO   

           STATUS LOCALISATION      DECOUPZONE                 ZONENAME  \
0      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
1      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
2      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
3      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
4      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
5      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
6      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
7      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
8      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
9      In Service    INTERIEUR      Grand-NORD      ZONE A - Grand-NORD   
51351  In Service    INTERIEUR       Grand-EST       ZONE C - Grand-EST   
51352  In Service    INTERIEUR       Grand-EST       ZONE C - Grand-EST   
51353  In Service    INTERIEUR       Grand-SUD       ZONE B - Grand-SUD   
51354  In Service    INTERIEUR     

In [77]:
joined_inner.drop_duplicates(inplace=True)

In [78]:
head=joined_inner.head(10)
tail=joined_inner.tail(10)
frames2=[head,tail]
head_tail=pd.concat(frames2)
head_tail

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0         sms      0  2012-05-07            1  8118c3aa53d6c7b   
6       Voice     92  2012-05-07            4  ada82842ff7cf09   
12        sms     25  2012-05-07            0  74189e127808747   
18      Voice     16  2012-05-07            0  2313f147b4452d5   
24      Voice     35  2012-05-07            1  f7c3a56c793594a   
30        sms     25  2012-05-07            2  f109dc2017dbc47   
36      Voice     19  2012-05-07            2  bf485bba7992871   
42      Voice     20  2012-05-07            0  752143fd1c190c4   
48      Voice     60  2012-05-07            3  a0169735987a325   
54      Voice     46  2012-05-07            4  3e7375dc5652ad9   
51328     sms      0  2012-05-09            1  69fde0b7b086a3d   
51334   Voice      5  2012-05-09            2  8ed06e1c217a84d   
51337   Voice     68  2012-05-09            1  50311def88b6487   
51341   Voice     22  2012-05-09            2  ec599df47190ce7   
51344     sms     50  2012-05-09            1  c9cd42f2e442bcd   
51347   Voice      0  2012-05-09            2  0f9c90a6ee44083   
51350     sms      0  2012-05-09            1  dd953c2ab18c1b7   
51353   Voice     47  2012-05-09            1  144d260d1961913   
51356   Voice     17  2012-05-09            4  22e493a92442b73   
51358     sms      0  2012-05-09            2  32e2e7c90a9b0e9   

       DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     SITE_ID  Hour  \
0      1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   
6      d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   
12     3704381e2e76259     19e2e     19e2e  1b5540c02d     0   
18     4e4bb5a13d08fb9     19e2e     19e2e  1b5540c02d     0   
24     b738d6d953faa78     19e2e     19e2e  1b5540c02d     0   
30     88dc03da34feaab     19e2e     19e2e  1b5540c02d     0   
36     48f86edd17fb544     19e2e     19e2e  1b5540c02d     0   
42     06cc661e13d94e3     19e2e     19e2e  1b5540c02d     0   
48     337de904c48ed5e     19e2e     19e2e  1b5540c02d     0   
54     59e2913d56ba036     19e2e     19e2e  1b5540c02d     0   
51328  e09e736e46c5c3d     19e2e     19e2e  cef3f76edc     0   
51334  b37b6bf7e011735     19e2e     19e2e  b95f55689e     0   
51337  8c1ee17130aa6b6     19e2e     19e2e  d519a3f223     0   
51341  d550086aa93330f     19e2e     19e2e  b8d2c54e65     0   
51344  6c64d047b6e8811     19e2e     19e2e  390877f878     0   
51347  1a49f8b0ded9de6     19e2e     19e2e  8cc89f3fa9     0   
51350  4b30ffe72e8a405     19e2e     19e2e  d236ec6ca8     0   
51353  fc15d238a9d904f     19e2e     19e2e  aa59cc7ced     0   
51356  c9e2c9c2fa3795a     19e2e     19e2e  cc45940406     0   
51358  32e2e7c90a9b0e9     19e2e     19e2e  ab91f5b21d     0   

                  VILLES      STATUS LOCALISATION      DECOUPZONE  \
0                 BOUAKE  In Service    INTERIEUR      Grand-NORD   
6                 BOUAKE  In Service    INTERIEUR      Grand-NORD   
12                BOUAKE  In Service    INTERIEUR      Grand-NORD   
18                BOUAKE  In Service    INTERIEUR      Grand-NORD   
24                BOUAKE  In Service    INTERIEUR      Grand-NORD   
30                BOUAKE  In Service    INTERIEUR      Grand-NORD   
36                BOUAKE  In Service    INTERIEUR      Grand-NORD   
42                BOUAKE  In Service    INTERIEUR      Grand-NORD   
48                BOUAKE  In Service    INTERIEUR      Grand-NORD   
54                BOUAKE  In Service    INTERIEUR      Grand-NORD   
51328  HERMANKONONO/DIES  In Service    INTERIEUR       Grand-SUD   
51334          DJASSANOU  In Service    INTERIEUR      Grand-NORD   
51337      BROU AKPASSOU  In Service    INTERIEUR       Grand-EST   
51341       MOUSSADOUGOU  In Service    INTERIEUR       SUD-OUEST   
51344            MBONOUA  In Service    INTERIEUR       Grand-SUD   
51347           PAKAOUBO  In Service    INTERIEUR      Grand-NORD   
51350            TIAPOUM  In Service    INTERIEUR       Grand-EST   
51353            BLANFLA  In Ser

In [79]:
joined_inner.shape

(8856, 19)

In [80]:
joined_inner.reset_index(drop=True,inplace=True)

In [81]:
head=joined_inner.head(10)
tail=joined_inner.tail(10)
frames2=[head,tail]
head_tail=pd.concat(frames2)
head_tail

PRODUCT  VALUE    DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0        sms      0  2012-05-07            1  8118c3aa53d6c7b   
1      Voice     92  2012-05-07            4  ada82842ff7cf09   
2        sms     25  2012-05-07            0  74189e127808747   
3      Voice     16  2012-05-07            0  2313f147b4452d5   
4      Voice     35  2012-05-07            1  f7c3a56c793594a   
5        sms     25  2012-05-07            2  f109dc2017dbc47   
6      Voice     19  2012-05-07            2  bf485bba7992871   
7      Voice     20  2012-05-07            0  752143fd1c190c4   
8      Voice     60  2012-05-07            3  a0169735987a325   
9      Voice     46  2012-05-07            4  3e7375dc5652ad9   
8846     sms      0  2012-05-09            1  69fde0b7b086a3d   
8847   Voice      5  2012-05-09            2  8ed06e1c217a84d   
8848   Voice     68  2012-05-09            1  50311def88b6487   
8849   Voice     22  2012-05-09            2  ec599df47190ce7   
8850     sms     50  2012-05-09            1  c9cd42f2e442bcd   
8851   Voice      0  2012-05-09            2  0f9c90a6ee44083   
8852     sms      0  2012-05-09            1  dd953c2ab18c1b7   
8853   Voice     47  2012-05-09            1  144d260d1961913   
8854   Voice     17  2012-05-09            4  22e493a92442b73   
8855     sms      0  2012-05-09            2  32e2e7c90a9b0e9   

      DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     SITE_ID  Hour  \
0     1d103ea638ad1f0     19e2e     19e2e  1b5540c02d     0   
1     d35102ff14c13b2     19e2e     19e2e  1b5540c02d     0   
2     3704381e2e76259     19e2e     19e2e  1b5540c02d     0   
3     4e4bb5a13d08fb9     19e2e     19e2e  1b5540c02d     0   
4     b738d6d953faa78     19e2e     19e2e  1b5540c02d     0   
5     88dc03da34feaab     19e2e     19e2e  1b5540c02d     0   
6     48f86edd17fb544     19e2e     19e2e  1b5540c02d     0   
7     06cc661e13d94e3     19e2e     19e2e  1b5540c02d     0   
8     337de904c48ed5e     19e2e     19e2e  1b5540c02d     0   
9     59e2913d56ba036     19e2e     19e2e  1b5540c02d     0   
8846  e09e736e46c5c3d     19e2e     19e2e  cef3f76edc     0   
8847  b37b6bf7e011735     19e2e     19e2e  b95f55689e     0   
8848  8c1ee17130aa6b6     19e2e     19e2e  d519a3f223     0   
8849  d550086aa93330f     19e2e     19e2e  b8d2c54e65     0   
8850  6c64d047b6e8811     19e2e     19e2e  390877f878     0   
8851  1a49f8b0ded9de6     19e2e     19e2e  8cc89f3fa9     0   
8852  4b30ffe72e8a405     19e2e     19e2e  d236ec6ca8     0   
8853  fc15d238a9d904f     19e2e     19e2e  aa59cc7ced     0   
8854  c9e2c9c2fa3795a     19e2e     19e2e  cc45940406     0   
8855  32e2e7c90a9b0e9     19e2e     19e2e  ab91f5b21d     0   

                 VILLES      STATUS LOCALISATION      DECOUPZONE  \
0                BOUAKE  In Service    INTERIEUR      Grand-NORD   
1                BOUAKE  In Service    INTERIEUR      Grand-NORD   
2                BOUAKE  In Service    INTERIEUR      Grand-NORD   
3                BOUAKE  In Service    INTERIEUR      Grand-NORD   
4                BOUAKE  In Service    INTERIEUR      Grand-NORD   
5                BOUAKE  In Service    INTERIEUR      Grand-NORD   
6                BOUAKE  In Service    INTERIEUR      Grand-NORD   
7                BOUAKE  In Service    INTERIEUR      Grand-NORD   
8                BOUAKE  In Service    INTERIEUR      Grand-NORD   
9                BOUAKE  In Service    INTERIEUR      Grand-NORD   
8846  HERMANKONONO/DIES  In Service    INTERIEUR       Grand-SUD   
8847          DJASSANOU  In Service    INTERIEUR      Grand-NORD   
8848      BROU AKPASSOU  In Service    INTERIEUR       Grand-EST   
8849       MOUSSADOUGOU  In Service    INTERIEUR       SUD-OUEST   
8850            MBONOUA  In Service    INTERIEUR       Grand-SUD   
8851           PAKAOUBO  In Service    INTERIEUR      Grand-NORD   
8852            TIAPOUM  In Service    INTERIEUR       Grand-EST   
8853            BLANFLA  In Service    INTERIEUR       Grand-SUD   
8854             ADJAME

**Most used cities for the three days**

In [82]:
joined_inner.groupby(['VILLES'])["VALUE"].sum().nlargest(5)

VILLES
YOPOUGON    39837
COCODY      36023
ABOBO       23260
KOUMASSI    13647
ADJAME      12933
Name: VALUE, dtype: int64

**Most used cities for each of the three days**

In [83]:
joined_inner.groupby(['DATETIME','VILLES'])["VALUE"].sum().nlargest(5)

DATETIME    VILLES  
2012-05-09  COCODY      16478
2012-05-07  YOPOUGON    14518
2012-05-09  YOPOUGON    13679
2012-05-08  YOPOUGON    11640
2012-05-07  COCODY      10787
Name: VALUE, dtype: int64

**Product use for the three days**

In [84]:
joined_inner.groupby(['PRODUCT','VILLES'])["VALUE"].sum()

PRODUCT  VILLES       
Voice    ABENGOUROU        1710
         ABOBO            18626
         ABOISSO            268
         ABOISSO COMOE      240
         ABONGOUA             1
                          ...  
sms      ZAIBO               50
         ZAROKO              25
         ZIKISSO            100
         ZOUKOUGBEU          50
         ZUENOULA           100
Name: VALUE, Length: 601, dtype: int64

In [85]:
joined_inner.groupby(['PRODUCT','VILLES'])["VALUE"].sum().nlargest(5)

PRODUCT  VILLES     
Voice    COCODY         32723
         YOPOUGON       32299
         ABOBO          18626
         TREICHVILLE    11748
         KOUMASSI       11472
Name: VALUE, dtype: int64